api/client/registry.go

```diff
@@ -91,6 +91,10 @@ func (cli *DockerCli) ConfigureAuth(flUser, flPassword, serverAddress string, is
 		cli.in = os.Stdin
 	}
 
+	if !isDefaultRegistry {
```

Comment:
Yeah, that's what I was thinking. I'm not 100% certain about this so I wonder if I'm missing something, but it seems like that would be more correct than relying on the `isDefaultRegistry` flag.